DevGuideExamples/FACE/two-topic/CMakeLists.txt

```diff
@@ -0,0 +1,45 @@
+cmake_minimum_required(VERSION 3.3...3.27)
+project(opendds_stock CXX)
+enable_testing()
+
+find_package(OpenDDS REQUIRED)
+include(opendds_testing)
+
+# Make sure the MPC-generated headers are gone so the CMake build will use the
+# right ones. This is not needed in a real project.
+# file(GLOB headers "*.h")
+# list(LENGTH headers header_count)
+# if(header_count GREATER 0)
+#   file(REMOVE ${headers})
+# endif()
+
+# IDL TypeSupport Library
+add_library(stockquoter_idl)
+include_directories(
+    /opt/OpenDDS/
+)
+opendds_target_sources(stockquoter_idl PUBLIC "StockQuoter.idl")
+target_link_libraries(stockquoter_idl PUBLIC OpenDDS::Dcps)
```

Comment:
link to `OpenDDS::FACE` before `opendds_target_sources` leads to an error at runtime: 

> subscriber-1  | free(): invalid pointer
subscriber-1 exited with code 139

Right now i don't if this is caused by the Subscriber code or other stuff.

Reply:
It needs to be before `opendds_target_sources` because otherwise it's using the normal IDL-to-C++ mapping, not the FACE one. Can you run it with gdb and post the trace?

Reply:
Nevermind, it was my problem.

Reply:
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007bc80b060859 in __GI_abort () at abort.c:79
#2  0x00007bc80b0cb26e in __libc_message (action=action@entry=do_abort, fmt=fmt@entry=0x7bc80b1f5298 "%s\n") at ../sysdeps/posix/libc_fatal.c:155
#3  0x00007bc80b0d32fc in malloc_printerr (str=str@entry=0x7bc80b1f34c1 "free(): invalid pointer") at malloc.c:5347
#4  0x00007bc80b0d4b2c in _int_free (av=<optimized out>, p=<optimized out>, have_lock=0) at malloc.c:4173
#5  0x00007bc80b57c393 in ACE_New_Allocator::free (this=0x7bc80b671e98 <ACE_Allocator::instance()::allocator_instance>, ptr=0x7bc80b884e64 <null_char>)
    at /opt/OpenDDS/ACE_wrappers/ace/Malloc_Allocator.cpp:127
#6  0x00007bc80d06c732 in FACE::string_free (str=0x7bc80b884e64 <null_char> "") at /opt/OpenDDS/dds/FACE/StringManager.cpp:37
#7  0x00005eb843246243 in OpenDDS::FaceTypes::StringTraits<char>::free (s=0x7bc80b884e64 <null_char> "") at /usr/local/include/FACE/StringManager.h:27
#8  0x00005eb843262281 in OpenDDS::FaceTypes::StringBase<char>::out (this=0x5eb844b782b0) at /usr/local/include/FACE/StringManager.h:92
#9  0x00005eb84323df2f in OpenDDS::DCPS::operator>> (strm=..., stru=...) at /opt/workspace/build/opendds_generated/telemetryTypeSupportImpl.cpp:1707
#10 0x00005eb8432426f0 in OpenDDS::DCPS::operator>> (strm=..., stru=...) at /opt/workspace/build/opendds_generated/telemetryTypeSupportImpl.cpp:3534
#11 0x00005eb84328a603 in OpenDDS::DCPS::DataReaderImpl_T<Telemetry::Message>::dds_demarshal (this=0x5eb844b75730, sample=..., publication_handle=5, 
    instance=..., just_registered=@0x7bc7fbffe080: false, filtered=@0x7bc7fbffe0f0: false, marshaling_type=OpenDDS::DCPS::FULL_MARSHALING)
    at /usr/local/include/dds/DCPS/DataReaderImpl_T.h:1141
#12 0x00007bc80c234908 in OpenDDS::DCPS::DataReaderImpl::data_received (this=0x5eb844b75888, sample=...) at /opt/OpenDDS/dds/DCPS/DataReaderImpl.cpp:1419
#13 0x00007bc80c5878a1 in OpenDDS::DCPS::ReceiveListenerSet::data_received (this=0x7bc7fc024800, sample=..., readerId=...)
    at /opt/OpenDDS/dds/DCPS/transport/framework/ReceiveListenerSet.cpp:141
#14 0x00007bc80c565426 in OpenDDS::DCPS::DataLink::data_received_i (this=0x7bc7fc0074e8, sample=..., readerId=..., Python Exception <class 'AttributeError'> 'NoneType' object has no attribute 'pointer': 
incl_excl=std::set with 0 elements, 
    constrain=OpenDDS::DCPS::ReceiveListenerSet::SET_EXCLUDED) at /opt/OpenDDS/dds/DCPS/transport/framework/DataLink.cpp:766
#15 0x00007bc80c564df9 in OpenDDS::DCPS::DataLink::data_received (this=0x7bc7fc0074e8, sample=..., readerId=...)
    at /opt/OpenDDS/dds/DCPS/transport/framework/DataLink.cpp:693
#16 0x00007bc80ac990a4 in OpenDDS::DCPS::RtpsUdpReceiveStrategy::deliver_sample_i (this=0x7bc7fc00c8d0, sample=..., submessage=..., remote_addr=...)
    at /opt/OpenDDS/dds/DCPS/transport/rtps_udp/RtpsUdpReceiveStrategy.cpp:604
#17 0x00007bc80ac98bab in OpenDDS::DCPS::RtpsUdpReceiveStrategy::deliver_sample (this=0x7bc7fc00c8d0, sample=..., remote_address=...)
    at /opt/OpenDDS/dds/DCPS/transport/rtps_udp/RtpsUdpReceiveStrategy.cpp:538
#18 0x00007bc80ac96e5a in OpenDDS::DCPS::RtpsUdpReceiveStrategy::handle_input (this=0x7bc7fc00c8d0, fd=11)
    at /opt/OpenDDS/dds/DCPS/transport/rtps_udp/RtpsUdpReceiveStrategy.cpp:169
#19 0x00007bc80c36339f in ACE_Select_Reactor_T<ACE_Reactor_Token_T<ACE_Token> >::notify_handle (this=0x5eb844b79df0, handle=11, mask=1, ready_mask=..., 
    event_handler=0x7bc7fc00cea8, ptmf=&virtual table offset 48) at /opt/OpenDDS/ACE_wrappers/ace/Select_Reactor_T.cpp:826
#20 0x00007bc80c363249 in ACE_Select_Reactor_T<ACE_Reactor_Token_T<ACE_Token> >::dispatch_io_set (this=0x5eb844b79df0, number_of_active_handles=1, 
    number_of_handlers_dispatched=@0x7bc7fbffeba8: 1, mask=1, dispatch_mask=..., ready_mask=..., callback=&virtual table offset 48)
    at /opt/OpenDDS/ACE_wrappers/ace/Select_Reactor_T.cpp:1210
#21 0x00007bc80c36312a in ACE_Select_Reactor_T<ACE_Reactor_Token_T<ACE_Token> >::dispatch_io_handlers (this=0x5eb844b79df0, dispatch_set=..., 
    number_of_active_handles=@0x7bc7fbffeb94: 1, number_of_handlers_dispatched=@0x7bc7fbffeba8: 1) at /opt/OpenDDS/ACE_wrappers/ace/Select_Reactor_T.cpp:1273
#22 0x00007bc80c362e64 in ACE_Select_Reactor_T<ACE_Reactor_Token_T<ACE_Token> >::dispatch (this=0x5eb844b79df0, active_handle_count=1, dispatch_set=...)
    at /opt/OpenDDS/ACE_wrappers/ace/Select_Reactor_T.cpp:1375
#23 0x00007bc80c366e32 in ACE_Select_Reactor_T<ACE_Reactor_Token_T<ACE_Token> >::handle_events_i (this=0x5eb844b79df0, max_wait_time=0x0)
    at /opt/OpenDDS/ACE_wrappers/ace/Select_Reactor_T.cpp:1462
#24 0x00007bc80c35fca3 in ACE_Select_Reactor_T<ACE_Reactor_Token_T<ACE_Token> >::handle_events (this=0x5eb844b79df0, max_wait_time=0x0)
    at /opt/OpenDDS/ACE_wrappers/ace/Select_Reactor_T.cpp:1439
#25 0x00007bc80b5c40e9 in ACE_Reactor::run_reactor_event_loop (this=0x5eb844b721e0, eh=0x0) at /opt/OpenDDS/ACE_wrappers/ace/Reactor.cpp:218
#26 0x00007bc80c35ab7f in OpenDDS::DCPS::ReactorTask::svc (this=0x5eb844b79c70) at /opt/OpenDDS/dds/DCPS/ReactorTask.cpp:171
#27 0x00007bc80b5ecd90 in ACE_Task_Base::svc_run (args=0x5eb844b79d40) at /opt/OpenDDS/ACE_wrappers/ace/Task.cpp:269
--Type <RET> for more, q to quit, c to continue without paging--
#28 0x00007bc80b5ed431 in ACE_Thread_Adapter::invoke_i (this=0x5eb844b84b40) at /opt/OpenDDS/ACE_wrappers/ace/Thread_Adapter.cpp:161
#29 0x00007bc80b5ed2c7 in ACE_Thread_Adapter::invoke (this=0x5eb844b84b40) at /opt/OpenDDS/ACE_wrappers/ace/Thread_Adapter.cpp:96
#30 0x00007bc80b52ea51 in ace_thread_adapter (args=0x5eb844b84b40) at /opt/OpenDDS/ACE_wrappers/ace/Base_Thread_Adapter.cpp:124
#31 0x00007bc80b238609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#32 0x00007bc80b15d353 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Reply:
About thath error, I found out that maybe this is happening because the docker image (the github one) is not set for FACE. I tried installing everything from scratch, but the docs didn't help me much.

However the docs say that the final part of the compilation is `make install` but if you specified FACE, target install is not present. About that, any help/clarification would be really appreciated.